api/builtins/builtins.go

```diff
@@ -0,0 +1,7 @@
+package builtins
+
+import (
+	internal "sigs.k8s.io/kustomize/api/internal/builtins"
+)
+
+var NewPrefixSuffixTransformerPlugin = internal.NewPrefixSuffixTransformerPlugin
```

Comment:
1. Mark as deprecated with a comment
2. Let's do the same for types, so add `type PrefixSuffixTransformerPlugin = internal.PrefixSuffixTransformerPlugin` (mark this deprecated too). 
3. Let's do this for all builtin plugins - not just PrefixSuffixTransformer, but all of the plugins in `api/builtins` - in case there are external consumers for other plugins. 

We will do similar aliasing for all tasks #3942 until they are all done, and then we can remove all the type aliases. The purpose is so that any external consumer of the api module won't see any changes until we are ready to release api v1.